articles/asdf-vim-plugin.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "asdf プラグインの作成 : Vimの場合"
+emoji: "🛠"
+type: "tech" # tech: 技術記事 / idea: アイデア
+topics: [vim, asdf]
+published: true
+---
+
+この記事は [asdf プラグインの作成 : Vimの場合 - Qiita](https://qiita.com/tsuyoshi_cho/items/495f081117253f0b23bb) と同じものです。
+
+----
+
+# asdf-vm プラグインの作成
+
+[asdf-vm](https://github.com/asdf-vm/asdf) はバージョニング管理ツール(でいいのかな?)で、[プラグイン](https://github.com/asdf-vm/asdf-plugins) による複数種のツールの管理、また同機構での拡張ができるのが特徴です。
+
+## asdf-vim プラグイン
+
+さて、このプラグインの一覧を見たときに、[Vim](https://github.com/vim/vim) がないことに気付きました。
+
+Vimはパッチバージョンの更新が頻繁で、リリースという明確な区切りがないので、ときどき不具合ありに当ってしまうことがあります。
+
+このときに回避したり、どこで混入したか調べたりするのに、古いバージョンを簡単に導入・切り替えできると嬉しいところがあります。
+
+(とはいえ、作った者は、普段は[Linuxbrew](https://docs.brew.sh/Homebrew-on-Linux)で更新していってるので、これであれこれすることは少なそうですが...)
+
+そこで、練習と環境充実を兼ねて[asdf-vim](https://github.com/tsuyoshicho/asdf-vim)を作成しました。
+
+## asdfのプラグインとして
+
+下で作成自体については詳細に説明しますが、asdfのプラグインがどんなことをすればいいのか、簡単に説明します。
+
+* bin/install、bin/list-all というコマンド(基本シェルスクリプトだが)が入ったリポジトリを作る(これは最低限)
+* bin/list-all はどうにかやって、対象ツールのインストールとして対応するバージョンリストを出力する
+* bin/installは引数で上のバージョン、インストール先などの情報をもらうので、それをもってインストール処理を行う
+* `asdf plugin-install hoge <git htttps path>` でプラグインをインストールして使う
+* asdf-pluginsへPRするには、
+    * CIテストの追加、GitHub Acitonsならインストールテストをしてくれるactionが用意されている
+    * CIテストがGreenであること
+    * pluginsディレクトリのhogeファイルにターゲットリポジトリを記載
+    * READMEに名前、リンク、Statusなどを入れ、更新
+    * これをローカルチェック
+    * PRのステータスもGreenであること
+
+    を確認しこれらをもってPRとして追加依頼をする
```

Comment:
**[textlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Line 45 sentence length(101) exceeds the maximum sentence length of 100.
Over 1 characters. (ja-technical-writing/sentence-length)